.github/CODEOWNERS

```diff
@@ -57,6 +57,7 @@ exporter/tencentcloudlogserviceexporter/             @open-telemetry/collector-c
 
 extension/asapauthextension                          @open-telemetry/collector-contrib-approvers @jamesmoessis @MovieStoreGuy
 extension/awsproxy/                                  @open-telemetry/collector-contrib-approvers @anuraaga @Aneurysm9 @mxiamxia
+extension/basicauth/                                 @open-telemetry/collector-contrib-approvers @jpkrohling @svrakitin
```

Comment:
The other auth extensions have "authextension" as the suffix. I think it would make sense to follow the pattern for this one as well.

Reply:
I assumed it is unnecessary, because it is already under the `extension` parent, but will rename for the sake of consistency.

Reply:
Renamed.